src/util/triplea/MapXMLCreator/PlayerAndAlliancesPanel.java

```diff
@@ -0,0 +1,282 @@
+package util.triplea.MapXMLCreator;
+
+import java.awt.Dimension;
+import java.awt.Font;
+import java.awt.GridBagConstraints;
+import java.awt.GridBagLayout;
+import java.awt.Insets;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.util.ArrayList;
+import java.util.TreeSet;
+
+import javax.swing.AbstractAction;
+import javax.swing.JButton;
+import javax.swing.JLabel;
+import javax.swing.JOptionPane;
+import javax.swing.JPanel;
+import javax.swing.SwingUtilities;
+
+/**
+ * 
+ * @author Erik von der Osten
+ * 
+ */
+public class PlayerAndAlliancesPanel extends DynamicRowsPanel {
+		
+	private TreeSet<String> m_alliances = new TreeSet<String>();	
+	
+	public PlayerAndAlliancesPanel(final JPanel stepActionPanel)
+	{
+		super(stepActionPanel);
+	}
+	
+	public static void layout(final MapXMLCreator mapXMLCreator, final JPanel stepActionPanel)
+	{
+		if (s_me == null || !(s_me instanceof PlayerAndAlliancesPanel))

```

Comment:
"s_me" I was expecting to be defined in this class as a static.. I was wrong. Since it's a static, you can fully qualify it with the full class name, then it'll be clear where it live. Since it can be null, should be of type of optional as well to avoid NPE. Next, the logic flow depending upon class type is a code smell as well. Like you could use two different variables to enumerate the class types possible, and then use teh variables distinctly from each other, rather than having one variable take on the value of potentially different class types that have different behavior (that is defined by the client, not by the class. This is polymorphism turned upside down, with data hiding the class decides the behavior, here we have the client that inspects the class type and then chooses for the class what behavior will happen. It's more comlpicated than that due to coupled logic, but that is the gist of it..)